### PR TITLE
pueue: add page

### LINKED ISSUE #5195: Page request: `pueue`
Would love to see a page for [pueue](https://github.com/Nukesor/pueue). I would document it, but I am fairly new to the program myself and am probably not qualified. They have a decent help page already that can probably be pretty easily translated: https://github.com/Nukesor/pueue#how-to-use-it

### PR DESCRIPTION
**Pull request by bl-ue**
Closes #5195
Closes Nukesor/pueue#172

- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

cc @jakevossen5 @Nukesor

### CODE REVIEW

**Comment by Nukesor** [comment]
Looks good to me :)
Thanks for writing this! I didn't find time to read into tldr yet :D

Mentioning the wik might be useful: https://github.com/Nukesor/pueue/wiki/Get-started

**Comment by bl-ue** [comment]
@Nukesor the wiki _is_ useful, but the README by itself is quite good 🙂 and I think that users who go to the readme will probably find their way to the wiki page. Besides, we generally like to link to just the repo :)

**Comment by jakevossen5** [comment]
Looks great to me! Thanks @bl-ue!

**Comment by Nukesor** [comment]
@bl-ue Sounds good to me :+1:

**Review by navarroaxel — APPROVED** [review]

File: pages/common/pueue.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# pueue
+
+> Pueue is a command-line task management tool for sequential and parallel execution of long-running tasks.
+> More information: <https://github.com/Nukesor/pueue>.
+
+- Add a command to the task list:
+
+`pueue add -- {{tr -u a-z A-Z}}`
```
Comment (sbrl) [inline pages/common/pueue.md:8]: Why `{{tr -u a-z A-Z}}` and not `{{command}}`? I think the latter would be more clear in this instance.

File: pages/common/pueue.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+- Create a task group:
+
+`pueue group --add {{group_name}}`
+
+- Kill a task:
+
+`pueue kill {{task_id}}`
+
+- Set maximum amount of parallel tasks (queued tasked are started as needed to meet this limit):
```
Comment (sbrl) [inline pages/common/pueue.md:30]: Perhaps this will read better?
```suggestion
- Set maximum amount of parallel tasks (queued tasks are started as needed to meet this limit):
```

File: pages/common/pueue.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+`pueue parallel {{number_of_parallel_tasks}}`
+
+- Edit the command line of a stopped tasked, in the default editor (as specified by `$EDITOR`):
```
Comment (sbrl) [inline pages/common/pueue.md:34]: Perhaps this will read better?
```suggestion
- Edit the command line of a stopped task in the default editor (as specified by `$EDITOR`):
```

**Review by sbrl — CHANGES_REQUESTED** [review]
Thanks for the new page! I've left some comments below for you to review.

File: pages/common/pueue.md (hunk 1 above)
Comment (bl-ue) [reply pages/common/pueue.md:8]: I picked that because it's easy to demonstrate the functionality of `pueue send` and `pueue logs`, as one operates on stdin and the other on stdout. `-u` ensures the output is always unbuffered, so the stdin would immediately be processed and sent to stdout.  Would you still like it changed?

File: pages/common/pueue.md (hunk 2 above)
Comment (bl-ue) [reply pages/common/pueue.md:30]: Not 'perhaps', 'definitely'!

File: pages/common/pueue.md (hunk 3 above)
Comment (bl-ue) [reply pages/common/pueue.md:34]: What??!?!? How could it be?? 😄

File: pages/common/pueue.md (hunk 1 above)
Comment (bl-ue) [reply pages/common/pueue.md:8]: It also demostrates command flags and multiple arguments.

Comment (jakevossen5) [reply pages/common/pueue.md:8]: I kinda agree with @sbrl here. I feel like it doesn't get the the point of why this tool exists, which I would say is running commands that take a long time in the background. Maybe a better example is `tar` if we don't want ``{{command}}``? I feel like `pueue send`'s description is clear enough that we don't need the command here be based around stdin.

Comment (bl-ue) [reply pages/common/pueue.md:8]: Hmm...I forgot to mention, another reason I picked `tr` is because it runs forever, until you stop it.  If you still think so, I'll change it.  Please tell me.

Comment (jakevossen5) [reply pages/common/pueue.md:8]: It's just my two cents, @Nukesor probably should have the final say as the creator of `pueue`

Comment (jakevossen5) [reply pages/common/pueue.md:8]: Just adding this, under the [A few possible applications](https://github.com/Nukesor/pueue#why-should-i-use-it) they are all commands that take a while. I feel like `tldr` should have examples that show off what the command is useful for but again, @Nukesor probably knows best.

Comment (sbrl) [reply pages/common/pueue.md:8]: I think `{{command}}` would be the best solution here, as it's the most neutral. But I do appreciate the point about using a long-running command.

Comment (Nukesor) [reply pages/common/pueue.md:8]: Thinking about it, `{{command}}` is probably better, since it indicates that any command can be added.

I didn't think that it's this hard to get as much information as possible into two lines of documentation, without confusing the user :D.

Especially, since adding commands in Pueue can actually be quite tricky/confusing, as soon if any characters need escaping.
Fittingly, there was an [issue](https://github.com/Nukesor/pueue/issues/179) just a few days ago, regarding escaping on `pueue add`.

I thought about adding quotation marks around the command `pueue add "{{command}}"`, but this might lead to more confusion than it might clarify things. What do you think about this?

Comment (bl-ue) [reply pages/common/pueue.md:8]: :+1: for `pueue add "{{command}}"`.  @sbrl?

Comment (she3o) [reply pages/common/pueue.md:8]: How about creating separate pages for subcommand (like with git: `git-add.md`, `git-clone.md`). some subcommands deserve their own pages (e.g. `pueue add`) while others don't (e.g. `pueue clean`). 

> Thinking about it, {{command}} is probably better, since it indicates that any command can be added.

If we have `pueue-add.md`. the first example could show the generic `pueue add -- {{command}}`...

> Just adding this, under the A few possible applications they are all commands that take a while. I feel like tldr should have examples that show off what the command is useful for 

Next examples would show something like:

```
# pueue add

> Enqueue a task for execution

- add {{command}} to the default queue.
 
`pueue add {{command}}`

- pass flags and arguments {{command}} while adding.
 
`pueue add -- {{command --Arg1 -f}}`

- add command to a [g]roup, see "tldr pueue-group" to operate groups.

`pueue add -g "CPU_intensive" -- ffmpeg -i input.mp4 frame_%d.png`

- add command with a [l]abel after a [d]elay has passed, see "tldr pueue-enqueue" for accepted formats.

`pueue add -l "compressing large file" -d "wednesday 10:30pm" "7z a compressed.7z large_file.xml"`
```
`pueue.md` would look like:

```
  Check pueue version:

      pueue --version

  Show general help and available subcommands:

      pueue --help

  Show more example on a pueue subcommand (like `add`, `parallel`, etc.):

      tldr pueue-subcommand
```
If you approve this I can add commits on top of `bl-ue/add-pueue` branch.

Comment (bl-ue) [reply pages/common/pueue.md:8]: @she3o that sounds great!  I'd love for you to finish this! However, I think it would be better for me to finish this PR (I'll go with `"{{command}}"` since that's the consensus) and you to open separate PR(s) for GitHub UI clarity and also so you get your own contributions ;)

Comment (bl-ue) [reply pages/common/pueue.md:8]: I'll go with "{{command}}" since that's the consensus
```suggestion
`pueue add -- "{{command}}"`
```

**Review by sbrl — APPROVED** [review]
Thanks, @bl-ue!
